src/machine/usb/descriptor/midi.go

```diff
@@ -0,0 +1,249 @@
+	TypeClassSpecific,
+	0x01,
+	0x00, //
+	0x01, //
+	0x41,
+	0x00,
+}
+
+var ClassSpecificMIDIHeader = ClassSpecificType{
+	data: classSpecificMIDIHeader[:],
+}
+
+const classSpecificMIDIInJackLen = 6
+
+var classSpecificMIDIInJack1 = [classSpecificMIDIInJackLen]byte{
+	classSpecificMIDIInJackLen,
+	TypeClassSpecific,
+	0x02, // MIDI In jack
+	0x01, // Jack type (embedded)
+	0x01, // Jack ID
+	0x00, // iJack
+}
+
+var ClassSpecificMIDIInJack1 = ClassSpecificType{
+	data: classSpecificMIDIInJack1[:],
+}
+
+var classSpecificMIDIInJack2 = [classSpecificMIDIInJackLen]byte{
+	classSpecificMIDIInJackLen,
+	TypeClassSpecific,
+	0x02, // MIDI In jack
+	0x02, // Jack type (external)
+	0x02, // Jack ID
+	0x00, // iJack
+}
+
+var ClassSpecificMIDIInJack2 = ClassSpecificType{
+	data: classSpecificMIDIInJack2[:],
+}
+
+const classSpecificMIDIOutJackLen = 9
+
+var classSpecificMIDIOutJack1 = [classSpecificMIDIOutJackLen]byte{
+	classSpecificMIDIOutJackLen,
+	TypeClassSpecific,
+	0x03, // MIDI Out jack
+	0x01, // Jack type (embedded)
+	0x03, // Jack ID
+	0x01, // number of input pins
+	0x02, // source ID
+	0x01, // source pin
+	0x00, // iJack
+}
+
+var ClassSpecificMIDIOutJack1 = ClassSpecificType{
+	data: classSpecificMIDIOutJack1[:],
+}
+
+var classSpecificMIDIOutJack2 = [classSpecificMIDIOutJackLen]byte{
+	classSpecificMIDIOutJackLen,
+	TypeClassSpecific,
+	0x03, // MIDI Out jack
+	0x02, // Jack type (external)
+	0x04, // Jack ID
+	0x01, // number of input pins
+	0x01, // source ID
+	0x01, // source pin
+	0x00, // iJack
+}
+
+var ClassSpecificMIDIOutJack2 = ClassSpecificType{
+	data: classSpecificMIDIOutJack2[:],
+}
+
+const classSpecificMIDIEndpointLen = 5
+
+var classSpecificMIDIOutEndpoint = [classSpecificMIDIEndpointLen]byte{
+	classSpecificMIDIEndpointLen,
+	TypeClassSpecific,
+	0x01, // CS endpoint
+	0x01, // number MIDI IN jacks
+	0x01, // assoc Jack ID
+}
+
+var ClassSpecificMIDIOutEndpoint = ClassSpecificType{
+	data: classSpecificMIDIOutEndpoint[:],
+}
+
+var classSpecificMIDIInEndpoint = [classSpecificMIDIEndpointLen]byte{
+	classSpecificMIDIEndpointLen,
+	TypeClassSpecific,
+	0x01, // CS endpoint
+	0x01, // number MIDI Out jacks
+	0x03, // assoc Jack ID
+}
+
+var ClassSpecificMIDIInEndpoint = ClassSpecificType{
+	data: classSpecificMIDIInEndpoint[:],
+}
+
+const endpointMIDITypeLen = 9
+
+var endpointEP6IN = [endpointMIDITypeLen]byte{
+	endpointMIDITypeLen,
+	TypeEndpoint,
+	0x86, // EndpointAddress
+	0x02, // Attributes
+	0x40, // MaxPacketSizeL
+	0x00, // MaxPacketSizeH
+	0x00, // Interval
+	0x00, // refresh
+	0x00, // sync address
+}
+
+var EndpointEP6IN = EndpointType{
+	data: endpointEP6IN[:],
+}
+
+var endpointEP7OUT = [endpointMIDITypeLen]byte{
+	endpointMIDITypeLen,
+	TypeEndpoint,
+	0x07, // EndpointAddress
+	0x02, // Attributes
+	0x40, // MaxPacketSizeL
+	0x00, // MaxPacketSizeH
+	0x00, // Interval
+	0x00, // refresh
+	0x00, // sync address
+}
+
+var EndpointEP7OUT = EndpointType{
+	data: endpointEP7OUT[:],
+}
```

Comment:
Shouldn't it be implemented in endpoint.go?

Reply:
I was thinking about that, however since these endpoints have values specific to the MIDI protocol (note the length difference).

Perhaps they should have `MIDI` in the endpoint name to make that more explicit?